Mathlib/Tactic/ExtendDoc.lean

```diff
@@ -26,18 +26,46 @@ At least one of `before` and `after` must appear, but either one of them is opti
 
 namespace Mathlib.Tactic.ExtendDocs
 
+open Lean Elab Command
+
+set_option autoImplicit true in
+open private docStringExt in addDocString in
+/-- Replace a declaration's docstring. If `declName` does not have a docstring yet, add it.
+
+Note that the docstring extension maintains both a `List (Name × String)` (for serialization) and a
+`NameMap String` (for state). Each `Name` in this `List` should be unique.
+`addDocString declName docString` assumes that `declName` does not have a docstring yet, and simply
+prepends `(declName, docString)` to the list. As such, it is unsuitable for modifying extant
+docstrings. -/
+private def replaceDocString [Monad m] [MonadError m] [MonadEnv m]
+    (declName : Name) (docString : String) : m Unit := do
+  unless (← getEnv).getModuleIdxFor? declName |>.isNone do
+    throwError s!"invalid doc string, declaration '{declName}' is in an imported module"
+  modifyEnv fun env => PersistentEnvExtension.modifyState docStringExt env fun (entries, map) =>
+    let entries := if map.contains declName then
+      modifyEntry entries else (declName, docString) :: entries
+    -- Note that `NameMap.insert` overwrites any existing key-value pairs.
+    (entries, map.insert declName docString)
+where
+  /-- Replace the `String` in the first pair that contains `declName`. (`declName` is assumed to
+  appear somewhere in the list.) -/
+  modifyEntry : List (Name × String) → List (Name × String)
+  | e@(n, _) :: entries =>
+    if n == declName then (n, docString) :: entries else e :: modifyEntry entries
+  | [] => []
+
+
```

Comment:
```suggestion

```
That's as far as I can give suggestions to this PR!  :smile:

Reply:
I'm not sure about what happened: I selected the *two* line breaks and removed *one* in the suggestion.  Maybe, I should have selected *one* line break and removed it?

Reply:
The source of your suggestion looks sensible! Hmm, maybe GitHub trims whitespace at a point it shouldn't? Or something? Either way, I'm blaming this on a bug... 😁